### ISSUE TITLE
Issue with Part-select in coverpoints 

### ISSUE THREAD

**Issue by ishita-71** [fvutils/pyvsc#97]
Hi mballance,
I tried with the fix you provided. for earlier test it passed but for the below case it is still giving same error

```import vsc
from vsc import *
     
@vsc.covergroup
class my_cg(object):
    def __init__(self):
         self.with_sample(a = vsc.bit_t(32))

         self.cp_my = vsc.coverpoint(lambda:self.a[7:2],
                                            bins={
                                                "a": vsc.bin_array([], [0, 3])
                                            }
                                            )
# Create an instance of the covergroup
my_cg_i = my_cg()
my_cg_i.sample(1)
```
output:
 `Traceback (most recent call last): File "example.py", line 24, in <module> my_cg_i.sample(1) File "/home/ishita/.local/lib/python3.6/site-packages/vsc/coverage.py", line 149, in sample model.sample() File "/home/ishita/.local/lib/python3.6/site-packages/vsc/model/covergroup_model.py", line 67, in sample cp.sample() File "/home/ishita/.local/lib/python3.6/site-packages/vsc/model/coverpoint_model.py", line 127, in sample b.sample() File "/home/ishita/.local/lib/python3.6/site-packages/vsc/model/coverpoint_bin_array_model.py", line 58, in sample val = int(self.cp.get_val()) File "/home/ishita/.local/lib/python3.6/site-packages/vsc/model/coverpoint_model.py", line 153, in get_val self.target_val_cache = self.target.val() File "/home/ishita/.local/lib/python3.6/site-packages/vsc/model/expr_ref_model.py", line 34, in val return self.ref() File "example.py", line 15, in <lambda> self.cp_my = vsc.coverpoint(lambda:self.a[7:2], TypeError: 'int' object is not subscriptable`
Can you help me with this?

**Comment by mballance** [comment]
Hi @ishita-71,
    Good point on this. In general, 'lamda' in a coverpoint expression is used to fetch a Python value. This does limit what PyVSC is able to do in terms of operator overloading (eg Python doesn't support part selects on scalar variables). Because you're referencing a PyVSC variable that is part of the covergroup in this case, it's better to reference the variable directly without a lamda.

```
        class my_cg(object):
            def __init__(self):
                self.with_sample(a = vsc.bit_t(32))
        
                self.cp_my = vsc.coverpoint(self.a[7:2],
                                                    bins={
                                                        "a": vsc.bin_array([], [0, 3])
                                                    }
                                                    )
        # Create an instance of the covergroup
        my_cg_i = my_cg()
        my_cg_i.sample(1)

```

After doing so, I did have to resolve a couple of issues with the new support for part-select expressions. These are resolved in the 0.4.9 release. 

With respect to #95, the lamba can be omitted in this case as well:
```
        import vsc
 
        @vsc.randobj
        class my_item_c(object):
            def __init__(self):
                self.a = vsc.bit_t(8)
    
        @vsc.covergroup
        class my_cg(object):
            def __init__(self):
                self.with_sample(dict(
                    it=my_item_c()))

                self.cp_my = vsc.coverpoint(self.it.a[7:2],
                                            bins={
                                                "a": vsc.bin_array([], [0, 3])
                                            }
                                            )
        # Create an instance of the covergroup
        my_cg_i = my_cg()
        # Create an instance of the item class
        my_item_i = my_item_c()
        my_cg_i.sample(my_item_i)    
```

Hopefully the 0.4.9 release resolves your issues with part selects. I'll leave the issue open until you can confirm.

Best Regards,
Matthew

**Comment by ishita-71** [comment]
Hi
